### PR TITLE
feat(internal/postprocessing): add removeFile function

### LINKED ISSUE #6296: java: create removeFile function in postprocessing package
Implement the declarative `removeFile` post-processing function in Go using `os.RemoveAll` to delete unwanted files.

See [go/sdk-librarian-mod-gen](https://docs.google.com/document/d/1a4R-tNZCbGN5TUTMRR6bG-CCDob1x0R7YDzmh5-9oC4/edit?resourcekey=0-XeSTzmtjPWsC6mmkpr0hMQ&tab=t.0#heading=h.lg3orppkodpi) for more details.

### PR DESCRIPTION
**Pull request by yangyzs**
Stacked PR: This depends on #6364. Please review and merge #6364 first. 

Add the RemoveFile function to the filesystem package and wrap it in the postprocessing package. It deletes specified files and silently ignores errors if the target file doesn't exist.

Fixes #6296

### CODE REVIEW

File: internal/filesystem/filesystem.go [hunk 1]
```diff
@@ -82,3 +84,13 @@ func CopyFile(src, dest string) error {
 func Unzip(ctx context.Context, src, dest string) error {
 	return command.Run(ctx, "unzip", "-q", "-o", src, "-d", dest)
 }
+
+// RemoveFile removes the file at the specified path.
+// It silently ignores errors if the file does not exist.
+func RemoveFile(path string) error {
```
Comment (JoeWang1127) [inline internal/filesystem/filesystem.go]: I think you don't need to create a function here, just implementing the function in postprocessing package.

File: internal/postprocessing/fileops_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestCopyFile_Success(t *testing.T) {
```
Comment (JoeWang1127) [inline internal/postprocessing/fileops_test.go]:
```suggestion
func TestCopyFile(t *testing.T) {
```

File: internal/postprocessing/fileops_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+func TestRemoveFile(t *testing.T) {
+	dir := t.TempDir()
+	path := filepath.Join(dir, "test.txt")
+	if err := os.WriteFile(path, []byte("content"), 0644); err != nil {
+		t.Fatal(err)
+	}
+	if err := RemoveFile(path); err != nil {
+		t.Fatal(err)
+	}
```
Comment (JoeWang1127) [inline internal/postprocessing/fileops_test.go:67]: Could you verify this file is no longer there?

File: internal/filesystem/filesystem.go (hunk 1 above)
Comment (yangyzs) [reply internal/filesystem/filesystem.go]: Moved the RemoveFile implementation into postprocessing/fileops.go .

File: internal/postprocessing/fileops_test.go (hunk 3 above)
Comment (yangyzs) [reply internal/postprocessing/fileops_test.go:67]: Verified actual file removal using os.Stat .

File: internal/postprocessing/fileops_test.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+func TestRemoveFile_NonExistent(t *testing.T) {
+	dir := t.TempDir()
+	path := filepath.Join(dir, "nonexistent.txt")
+	if err := RemoveFile(path); err != nil {
+		t.Fatalf("RemoveFile() returned unexpected error for non-existent file: %v", err)
+	}
+}
+
+func TestRemoveFile_Error(t *testing.T) {
```
Comment (JoeWang1127) [inline internal/postprocessing/fileops_test.go:81]: How about combine `TestRemoveFile_NonExistent` with this test to a table driven tests.

Comment (yangyzs) [reply internal/postprocessing/fileops_test.go:81]: Unified the error boundary cases into a Table-Driven TestRemoveFile_Error.

File: internal/postprocessing/fileops_test.go [hunk 5]
```diff
@@ -54,3 +54,56 @@ func TestCopyFile_Error(t *testing.T) {
...
+			if test.createDir {
+				if err := os.Mkdir(path, 0755); err != nil {
+					t.Fatal(err)
+				}
+				if err := os.WriteFile(filepath.Join(path, "sub.txt"), []byte("data"), 0644); err != nil {
+					t.Fatal(err)
+				}
+			}
+			err := RemoveFile(path)
+			if test.expectedError && err == nil {
```
Comment (JoeWang1127) [inline internal/postprocessing/fileops_test.go]: Could you check the error: https://github.com/googleapis/librarian/blob/main/doc/howwewritego.md#verify-with-errorsis-and-errorsas

File: internal/postprocessing/fileops_test.go [hunk 6]
```diff
@@ -54,3 +54,56 @@ func TestCopyFile_Error(t *testing.T) {
...
+}
+
+func TestRemoveFile_Error(t *testing.T) {
+	for _, test := range []struct {
+		name          string
+		createDir     bool
+		expectedError bool
+	}{
+		{
+			name:          "success non-existent file",
```
Comment (JoeWang1127) [inline internal/postprocessing/fileops_test.go]: Hmm, this test doesn't return an error, maybe we should create a new test. Then we only have one test to verify the error case. We don't need table driven test in this case.

File: internal/postprocessing/fileops_test.go (hunk 5 above)
Comment (yangyzs) [reply internal/postprocessing/fileops_test.go]: Done.

File: internal/postprocessing/fileops_test.go (hunk 6 above)
Comment (yangyzs) [reply internal/postprocessing/fileops_test.go]: Done. I've split this into 2 tests.

**Review by JoeWang1127 — APPROVED** [review]
